dipy/viz/regtools.py

```diff
@@ -90,162 +93,171 @@ def draw_lattice_2d(nrows, ncols, delta):
         if direct_grid_shape is None:
             direct_grid_shape = mapping.domain_shape
         if direct_grid_affine is -1:
             direct_grid_affine = mapping.domain_affine
 
-        #By default, the output grid is the mapping's domain grid
+        # By default, the output grid is the mapping's domain grid
         if inverse_grid_shape is None:
             inverse_grid_shape = mapping.codomain_shape
         if inverse_grid_affine is -1:
             inverse_grid_affine = mapping.codomain_affine
 
-    #The world-to-image (image = drawn lattice on the output grid)
-    #transformation is the inverse of the output affine
+    # The world-to-image (image = drawn lattice on the output grid)
+    # transformation is the inverse of the output affine
     world_to_image = None
     if inverse_grid_affine is not None:
-        world_to_image = np.linalg.inv(inverse_grid_affine) 
+        world_to_image = np.linalg.inv(inverse_grid_affine)
 
-    #Draw the squares on the output grid
-    X1,X0 = np.mgrid[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
-    lattice_out=draw_lattice_2d((inverse_grid_shape[0] + delta) / (delta + 1), 
-                                (inverse_grid_shape[1] + delta) / (delta + 1),
-                                delta)
-    lattice_out=lattice_out[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
+    # Draw the squares on the output grid
+    lattice_out = draw_lattice_2d(
+        (inverse_grid_shape[0] + delta) // (delta + 1),
+        (inverse_grid_shape[1] + delta) // (delta + 1),
+        delta)
+    lattice_out = lattice_out[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
 
-    #Warp in the forward direction (sampling it on the input grid)
+    # Warp in the forward direction (sampling it on the input grid)
     warped_forward = mapping.transform(lattice_out, 'linear', world_to_image,
                                        direct_grid_shape, direct_grid_affine)
 
-    
-    #Now, the world-to-image (image = drawn lattice on the input grid) 
-    #transformation is the inverse of the input affine
-    world_to_image = None 
+    # Now, the world-to-image (image = drawn lattice on the input grid)
+    # transformation is the inverse of the input affine
+    world_to_image = None
     if direct_grid_affine is not None:
-        world_to_image = np.linalg.inv(direct_grid_affine) 
+        world_to_image = np.linalg.inv(direct_grid_affine)
 
-    #Draw the squares on the input grid
-    X1,X0 = np.mgrid[0:direct_grid_shape[0], 0:direct_grid_shape[1]]
-    lattice_in=draw_lattice_2d((direct_grid_shape[0] + delta) / (delta + 1), 
-                               (direct_grid_shape[1] + delta) / (delta + 1),
-                               delta)
-    lattice_in=lattice_in[0:direct_grid_shape[0], 0:direct_grid_shape[1]]
+    # Draw the squares on the input grid
+    lattice_in = draw_lattice_2d((direct_grid_shape[0] + delta) // (delta + 1),
```

Comment:
Yes, it is safer not to deal with floating point operations when possible (we might get different results depending on the precision of the floats being used, like the double vs. extended discussion we had some time ago)